akka-stream-tests/src/test/scala/akka/stream/scaladsl/BidiFlowSpec.scala

```diff
@@ -121,6 +119,43 @@ class BidiFlowSpec extends StreamSpec {
       b.traversalBuilder.attributes.getFirst[AsyncBoundary.type] shouldEqual Some(AsyncBoundary)
     }
 
+    "short circuit identity in atop" in {
+      val myBidi = BidiFlow.fromFlows(Flow[Long].map(_ + 1L), Flow[ByteString])
+      val identity = BidiFlow.identity[Long, ByteString]
+
+      // simple ones
+      myBidi.atop(identity) should ===(myBidi)
+      identity.atopMat(myBidi)(Keep.right) should ===(myBidi)
+
+      // trickier because implicit Keep.left means keep NotUsed matval from identity
+      // optimized but not the same instance
+      identity.atop(myBidi) should !==(myBidi)
```

Comment:
I think for those cases we need more assertions to actually test what it is and not only what it is not:

 1. check traversal that it doesn't contain any identity stages
 2. either check complete traversal that it is what we expect, or check behavior (also regarding attributes, island behavior, etc.)

Reply:
Ah, I guess that's done below?

Reply:
Well, at least the tests below that it works when materialized :)